stacks/caip10.md

```diff
@@ -0,0 +1,57 @@
+---
+namespace-identifier: stacks-caip10
+title: Stacks Namespace - Addresses
+author: Léo Pradel @pradel, Greg Ogunyanwo @gregogun
+discussions-to: https://forum.stacks.org/t/caip-2-and-caip-10-stacks-specification/13290
+status: Draft
+type: Standard
+created: 2022-06-07
+requires: "CAIP-10"
+---
+
+# CAIP-10
+
+*For context, see the [CAIP-10][] specification.*
+
+## Rationale
+
+Stacks aligns squarely with the "account" model rather than the "UXTO" model but
+uses the format "c32check" to address the "account" model.
+
+## Syntax
+
+Address string concatenates 2 fields without whitespace or dividers: a 1-byte version prefix and a 20-byte hash of any one of the following:
```

Comment:
Is this a complete list? Could it be a taproot script as well?

Reply:
@pradel ? I'm a way out of my depth here, feel I can't quite make this call for myself.  I'm judging from this:
https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#taproot-key-path-spending-signature-validation
that probably P2TR addresses would probably be difficult to "sniff" or detect if 64-byte and 65-byte addresses both matched the regEx you added... maybe say either way whether all of this DOES or DOESN'T work with P2TR addresses in the body text?  Other than this one nit it looks complete enough to merge IMHO, although admittedly out of my depth so I'm putting outsized faith in the secondary review of @friedger :D

Reply:
@bumblefudge The stx address is a hash therefore the source of the hash can be anything, the regex will match. 

This caip10 limits the source of the hash to a list. This list should include p2tr. @pradel

Reply:
@friedger @bumblefudge updated to add p2tr

Reply:
@pradel apparently my proposal was wrong. Could you please revert the last change? Then it is good to merge.

Reply:
Wait @friedger what do you mean wrong? I just tried googling a little harder and at least according to [this medium changelog from august](https://stacks.org/stacks-21-what-to-expect) taproot addresses work exactly the same for these purposes.  so i think your proposal was... very good? I wasn't trying to imply otherwise, just telling Pradel he should make the call, not me, as I barely know what Stacks ID :D.  

In any case, thanks so much for the review, and hope the CAIPs are useful to your work!